src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+        if (type.equals(Excavator.class)) {
+            MachineProducer<Excavator> producer = new ExcavatorProducer();
+            machines.addAll(producer.get());
+        } else if (type.equals(Bulldozer.class)) {
+            MachineProducer<Bulldozer> producer = new BulldozerProducer();
+            machines.addAll(producer.get());
+        } else if (type.equals(Truck.class)) {
+            MachineProducer<Truck> producer = new TruckProducer();
+            machines.addAll(producer.get());
+        }
+        return machines;
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method to store the result from `producer.get()` and then return a new ArrayList with those elements. This would make the code cleaner and ensure that you're not modifying the original list returned by `producer.get()`.